client/config.c

```diff
@@ -91,6 +141,9 @@ TDNFReadConfig(
         }
     }
 
+    dwError = TDNFParseOSInfo(pConf);
+    BAIL_ON_TDNF_ERROR(dwError);
```

Comment:
We should not bail here if the error is that the file wasn't found. In that case put "unknown" as the OS and version.